Lib/unittest/test/testmock/testmock.py

```diff
@@ -1808,6 +1811,29 @@ def test_attach_mock_return_value(self):
                 self.assertEqual(m.mock_calls, call().foo().call_list())
 
 
+    def test_attach_mock_patch_autospec(self):
+        # bpo-21478: attach_mock used with autospecced object should have
+        # should use child's name and record calls to parent when being called
+        # as child and also as standalone functions.
```

Comment:
Not sure we need this comment, but the text could do with some tweaking.

Reply:
I removed it since I feel bug report gives the relevant context. Feel free to suggest wording if I need to incorporate a code comment. Since I am not a native speaker sometimes my language might be little difficult to grasp.